Stack/Core/Stack/State/NodeState.cs

```diff
@@ -201,25 +207,32 @@ public string ToString(string format, IFormatProvider formatProvider)
         public virtual object Clone()
         {
             NodeState clone = (NodeState)this.MemberwiseClone();
-
-            if (m_children != null)
+            lock (m_childrenLock)
             {
-                clone.m_children = new List<BaseInstanceState>(m_children.Count);
-
-                for (int ii = 0; ii < m_children.Count; ii++)
+                if (m_children != null)
                 {
-                    BaseInstanceState child = (BaseInstanceState)m_children[ii].Clone();
-                    clone.m_children.Add(child);
+                    clone.m_children = new List<BaseInstanceState>(m_children.Count);
+
+                    for (int ii = 0; ii < m_children.Count; ii++)
+                    {
+                        BaseInstanceState child = (BaseInstanceState)m_children[ii].Clone();
+                        clone.m_children.Add(child);
+                    }
                 }
             }
-
             clone.m_changeMasks = NodeStateChangeMasks.None;
 
             return clone;
         }
         #endregion
 
         #region Public Properties
+
+        /// <summary>
+        /// Indicates whether Bad StatusCode is returned when ServerTimestamp is written to non value attributes.
```

Comment:
Lets fix the comment to "...BadWriteNotSupported is returned when SourceTimestamp or ServerTimestamp..."